### PR TITLE
Import assets from markdown file

### PR DESCRIPTION
**Pull request by Mukund-Tandon**
This PR adds support to detect images from markdown file which are in Alt text format
It also fixes the error where if a file is there in markdown file it is detected as a url since both have a similar representation in markdown - [Alt tex](url or file path) not if it is a file it is treated as a simple paragraph  with text "This is currently not supported : [Alt text](file path)"
Related to https://github.com/AppFlowy-IO/AppFlowy/pull/2981

### CODE REVIEW

**Comment by Mukund-Tandon** [comment]
@LucasXu0 ,please can you review this PR

File: lib/src/plugins/markdown/decoder/document_markdown_decoder.dart [hunk 1]
```diff
@@ -52,7 +52,9 @@ class DocumentMarkdownDecoder extends Converter<String, Document> {
 
   Node _convertLineToNode(String line) {
     final decoder = DeltaMarkdownDecoder();
-
+    final imageRegex = RegExp(r'^!\[.*\]\(.*\)$');
+    final assetRegex = RegExp(r'^\[.*\]\(.*\)$');
+    final htmlRegex = RegExp('^(http|https)://');
```
Comment (LucasXu0) [inline lib/src/plugins/markdown/decoder/document_markdown_decoder.dart:57]: Define them as class properties, eliminating the need to create the regex every time.

File: lib/src/plugins/markdown/decoder/document_markdown_decoder.dart [hunk 2]
```diff
@@ -91,6 +93,31 @@ class DocumentMarkdownDecoder extends Converter<String, Document> {
...
+        );
+      } else {
+        return imageNode(url: filepath);
+      }
+    } else if (assetRegex.hasMatch(line.trim())) {
+      //this might be a url or a file like pdf,videos,etc
+      String? filepath = extractFilePath(line.trim());
+      if (filepath != null && !htmlRegex.hasMatch(filepath)) {
+        return paragraphNode(
+          text: "This file is cuurently not supported : $line",
```
Comment (LucasXu0) [inline lib/src/plugins/markdown/decoder/document_markdown_decoder.dart:114]: currently

**Comment by LucasXu0** [comment]
> @LucasXu0 ,please can you review this PR

I did some changes.

- use regex to match the image path and file path.
- remove the hint text(... not supported...).
- define the regexes as class properties.

**Comment by Mukund-Tandon** [comment]
> > @LucasXu0 ,please can you review this PR
> 
> @Mukund-Tandon I did some changes.
> 
> * use regex to match the image path and file path.
> * remove the hint text(... not supported...).
> * define the regexes as class properties.

Ok 👍

**Comment by Mukund-Tandon** [comment]
@LucasXu0 , Is there anything else I need to do?

**Comment by LucasXu0** [comment]
> @LucasXu0 , Is there anything else I need to do?

No, you can integrate the latest version into AppFlowy and continue developing the markdown import feature.

**Pull request merged**